catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/AbstractLocator.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard;
+
+import ddf.catalog.Constants;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.activation.MimeType;
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+
+public abstract class AbstractLocator {
+
+  <T extends ServiceProperties> boolean filterByMimeType(T transformer, MimeType userMimeType) {
+    return filterByMimeType(transformer.getMimeTypes(), transformer.getProperties(), userMimeType);
+  }
+
+  private boolean filterByMimeType(
+      Set<MimeType> mimeTypes, Map<String, Object> serviceProperties, MimeType userMimeType) {
+
+    Set<String> baseTypes =
+        mimeTypes.stream().map(MimeType::getBaseType).collect(Collectors.toSet());
+
+    String userIdValue = userMimeType.getParameter(Constants.SERVICE_ID);
+
+    String userBaseType = userMimeType.getBaseType();
+
+    String serviceId = getServiceId(serviceProperties);
+
+    return baseTypes.contains(userBaseType)
+        && (userIdValue == null || StringUtils.equals(userIdValue, serviceId));
+  }
+
+  private String getServiceId(Map<String, Object> properties) {
+    Object idServiceProperty = properties.get(Constants.SERVICE_ID);
+
+    if (idServiceProperty != null) {
+      return idServiceProperty.toString();
+    }
+
+    return null;
+  }
+
+  private BundleContext getContext() {
+    Bundle bundle = FrameworkUtil.getBundle(AbstractLocator.class);
+    if (bundle != null) {
+      return bundle.getBundleContext();
+    }
+    return null;
+  }
+
+  private <T> Collection<ServiceReference<T>> findServices(
+      Class<T> clazz, String filter, BundleContext bundleContext) {
+    try {
+      return bundleContext.getServiceReferences(clazz, filter);
+    } catch (InvalidSyntaxException e) {
+      throw new IllegalArgumentException("Invalid syntax supplied");
```

Comment:
Consider adding the filter that caused the exception

Reply:
done